shared/fs/container.js

```diff
@@ -75,6 +75,7 @@ const FilesLoadingHoc = compose(
       // This gets called on route changes too, e.g. when user clicks the
       // action menu. So only load folder list when path changes.
       this.props.path !== prevProps.path && this.props.loadFolderList(this.props.path)
+      Types.getPathLevel(this.props.path) === 2 && this.props.loadFavorites()
```

Comment:
We should still put this under the guard on `path` above right? Otherwise it gets fired when user clicks the action menu too.

Reply:
That's fine, I think. Ideally we'd like all favorites to be loaded when the component first loads so there's no major updates when they arrive at a favorites list.

Reply:
Or maybe I don't understand your suggestion.

Reply:
Oh what I meant was, instead of this:

```
this.props.path !== prevProps.path && this.props.loadFolderList(this.props.path)
Types.getPathLevel(this.props.path) === 2 && this.props.loadFavorites()
```

we have this:

```
```
if (this.props.path !== prevProps.path) {
  this.props.loadFolderList(this.props.path)
  Types.getPathLevel(this.props.path) === 2 && this.props.loadFavorites()
}
```

Because this gets fired even when user clicks an action menu, in which case we shouldn't need to reload.

Reply:
OK I'll change it.

Reply:
Thanks John!